benchmarks/source/client_benchmarks.cpp

```diff
@@ -23,58 +26,116 @@ namespace
-
-BENCHMARK(benchmark_client_send_event<Client>)->Iterations(number_of_iterations)->Repetitions(32);
-BENCHMARK(benchmark_client_send_metric<AsioClient>)->Iterations(number_of_iterations)->Repetitions(32);
-BENCHMARK(benchmark_client_send_event<SPSCClient>)->Iterations(number_of_iterations)->Repetitions(32);
-BENCHMARK(benchmark_client_send_event<MPMCClient>)->Iterations(number_of_iterations)->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<Client<transports::UDPTransport>, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<Client<transports::UDSTransport>, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<Client<transports::AsyncUDPTransport>, transports::AsyncUDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<Client<transports::AsyncUDSTransport>, transports::AsyncUDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+
+BENCHMARK(benchmark_client_send_metric<SPSCClient, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<SPSCClient, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+
+BENCHMARK(benchmark_client_send_metric<MPMCClient, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_metric<MPMCClient, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+
+BENCHMARK(benchmark_client_send_event<Client<transports::UDPTransport>, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_event<Client<transports::UDSTransport>, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_event<Client<transports::AsyncUDPTransport>, transports::AsyncUDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_event<Client<transports::AsyncUDSTransport>, transports::AsyncUDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+
+BENCHMARK(benchmark_client_send_event<SPSCClient, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_event<SPSCClient, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+
+BENCHMARK(benchmark_client_send_event<MPMCClient, transports::UDPTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
+BENCHMARK(benchmark_client_send_event<MPMCClient, transports::UDSTransport>)
+    ->Iterations(number_of_iterations)
+    ->Repetitions(32);
```

Comment:
I think its fine to have removed the allocation part of the benchmarks, but I think there's value in having a benchmark which is more alike regular use (creation and send-off)